csharp/lib/src/lib.rs

```diff
@@ -124,10 +218,18 @@ pub extern "C" fn set(
     });
 }
 
-/// Expects that key will be kept valid until the callback is called. If the callback is called with a string pointer, the pointer must
-/// be used synchronously, because the string will be dropped after the callback.
+/// Execute `GET` command. See [`redis.io`](https://redis.io/commands/get/) for details.
+///
+/// # Safety
+///
+/// * `client_ptr` must not be null.
+/// * `client_ptr` must be able to be safely casted to a valid `Box<Client>` via `Box::from_raw`. See the safety documentation of [`std::boxed::Box::from_raw`](https://doc.rust-lang.org/std/boxed/struct.Box.html#method.from_raw).
+/// * `key` must not be null.
+/// * `key` must be able to be safely casted to a valid `CString` via `CString::from_raw`. See the safety documentation of [`std::ffi::CString::from_raw`](https://doc.rust-lang.org/std/ffi/struct.CString.html#method.from_raw).
```

Comment:
Again, here you refer to CString but the implementation is using CStr, which is a different type and has different safety conditions.